cura/Settings/PerObjectContainerStack.py

```diff
@@ -51,3 +51,16 @@ def getProperty(self, key: str, property_name: str, context: Optional[PropertyEv
         result = super().getProperty(key, property_name, context)
         context.popContainer()
         return result
+
+    @override(ContainerStack)
+    def setNextStack(self, stack: ContainerStack):
+        super().setNextStack(stack)
+
+        # trigger signal to re-evaluate all default settings
+        for key, instance in self.getContainer(0)._instances.items():
+            # only evaluate default settings
+            if instance.state != InstanceState.Default:
+                continue
+
+            self._collectPropertyChanges(key, "value")
+        self._emitCollectedPropertyChanges()
```

Comment:
Settings on the stack should not get updated. They are fixed from the moment that they are added, and should not get re-evaluated after that.